be/src/vec/functions/array/function_array_apply.cpp

```diff
@@ -0,0 +1,220 @@
+        const auto* src_nested_column = &src_column_array->get_data();
+        DCHECK(src_nested_column != nullptr);
+
+        DataTypePtr src_column_type = block.get_by_position(arguments[0]).type;
+        auto nested_type = assert_cast<const DataTypeArray&>(*src_column_type).get_nested_type();
+        const std::string& condition =
+                block.get_by_position(arguments[1]).column->get_data_at(0).to_string();
+        if (!is_column_const(*block.get_by_position(arguments[2]).column)) {
+            return Status::RuntimeError(
+                    "execute failed or unsupported column, only support const column");
+        }
+        const ColumnConst& rhs_value_column =
+                static_cast<const ColumnConst&>(*block.get_by_position(arguments[2]).column.get());
+        ColumnPtr result_ptr;
+        RETURN_IF_ERROR(_execute(*src_nested_column, nested_type, src_offsets, condition,
+                                 rhs_value_column, &result_ptr));
+        block.replace_by_position(result, std::move(result_ptr));
+        return Status::OK();
+    }
+
+private:
+    enum class ApplyOp {
+        UNKNOWN = 0,
+        EQ = 1,
+        NE = 2,
+        LT = 3,
+        LE = 4,
+        GT = 5,
+        GE = 6,
+    };
+    template <typename T, ApplyOp op>
+    bool apply(T data, T comp) {
+        if constexpr (op == ApplyOp::EQ) {
+            return data == comp;
+        }
+        if constexpr (op == ApplyOp::NE) {
+            return data != comp;
+        }
+        if constexpr (op == ApplyOp::LT) {
+            return data < comp;
+        }
+        if constexpr (op == ApplyOp::LE) {
+            return data <= comp;
+        }
+        if constexpr (op == ApplyOp::GT) {
+            return data > comp;
+        }
+        if constexpr (op == ApplyOp::GE) {
+            return data >= comp;
+        }
+        __builtin_unreachable();
+    }
+
+    template <typename T, ApplyOp op>
+    ColumnPtr _apply_internal(const IColumn& src_column, const ColumnArray::Offsets64& src_offsets,
+                              const ColumnConst& cmp) {
+        T rhs_val = *reinterpret_cast<const T*>(cmp.get_data_at(0).data);
+        auto column_filter = ColumnUInt8::create(src_column.size(), 0);
+        auto& column_filter_data = column_filter->get_data();
+        const char* src_column_data_ptr = nullptr;
+        if (!src_column.is_nullable()) {
+            src_column_data_ptr = src_column.get_raw_data().data;
+        } else {
+            src_column_data_ptr = check_and_get_column<ColumnNullable>(src_column)
+                                          ->get_nested_column()
+                                          .get_raw_data()
+                                          .data;
+        }
+        for (size_t i = 0; i < src_column.size(); ++i) {
+            T lhs_val = *reinterpret_cast<const T*>(src_column_data_ptr);
+            if (apply<T, op>(lhs_val, rhs_val)) {
+                column_filter_data[i] = 1;
```

Comment:
There is some enhancement space, dst offset can be calculated in here.

Reply:
OK， but i the code will be little hard to understand to put offset caculated here, so i seperate them, I'll modify this if there is a performance issue